app/forms/idv/inherited_proofing/va/form.rb

```diff
@@ -0,0 +1,145 @@
+module Idv
+  module InheritedProofing
+    module Va
+      class Form
+        include ActiveModel::Model
+
+        class << self
+          def model_name
+            ActiveModel::Name.new(self, nil, namespaced_model_name)
+          end
+
+          def namespaced_model_name
+            self.to_s.gsub('::', '')
+          end
+
+          # Returns the field names based on the validators we've set up.
+          def field_names
+            @field_names ||= validators.filter_map do |validator|
+              validator.attributes.first
+            end&.flatten&.uniq&.sort
+          end
+        end
+
+        private_class_method :namespaced_model_name
+
+        attr_reader :payload_hash
+
+        validate :validate_field_names
+
+        validates :first_name, presence: true, length: { maximum: 255 }
+        validates :last_name, presence: true, length: { maximum: 255 }
+        validates :phone, length: { maximum: 11 }
+        validates :birth_date, presence: true
+        validates :ssn, presence: true
+        validates :address_street, presence: true, length: { maximum: 255 }
+        validates :address_street2, length: { maximum: 255 }
+        validates :address_city, length: { maximum: 255 }
+        validates :address_state, length: { maximum: 2 }
+        validates :address_country, length: { maximum: 255 }
+        validates :address_zip, presence: true
+        validates_format_of :address_zip, with: /\A\d{5}(-?\d{4})?\z/,
+                                          message: I18n.t('idv.errors.pattern_mismatch.zipcode'),
+                                          allow_blank: false
+
+        # This must be performed after our validators are defined.
+        attr_accessor(*self.field_names)
+
+        def initialize(payload_hash:)
+          raise ArgumentError, 'payload_hash is blank?' if payload_hash.blank?
+          raise ArgumentError, 'payload_hash is not a Hash' unless payload_hash.is_a? Hash
+
+          @payload_hash = payload_hash.dup
+
+          populate_field_data
+        end
+
+        def submit
+          validate
+
+          FormResponse.new(
+            success: valid?,
+            errors: errors,
+            extra: {
+            },
+          )
+        end
+
+        private
+
+        attr_writer :payload_hash
+
+        # Populates our field data from the payload hash.
+        def populate_field_data
+          payload_field_info.each do |field_name, field_info|
+            value = payload_hash.dig(*[field_info[:in], field_info[:field_name]].flatten.compact)
+            public_send("#{field_name}=", value)
+          end
+        end
+
+        # Validator for field names. All fields (not the presence of data) are required.
+        def validate_field_names
+          self.class.field_names.each do |field_name|
+            next if payload_field_info.key? field_name
+            errors.add(field_name, 'field is missing', type: missing_required_field)
+          end
+        end
```

Comment:
@holytoastr just ignoring any _extra_ fields delivered with the payload; I assumed we'd just ignore them. Thoughts?

Reply:
Makes sense to me. Ignoring what we don't need should do the job!